src/adapter.js

```diff
@@ -255,13 +308,12 @@ class MongooseAdapter {
         }
       }
 
-      await CasbinRule.collection.insertMany(lines, session ? { session: session } : null);
-      session && await session.commitTransaction();
+      await CasbinRule.collection.insertMany(lines, options);
     } catch (err) {
-      session && await session.abortTransaction();
+      options.session && await options.session.abortTransaction();
       throw err;
     } finally {
-      session && await session.endSession();
+      if (options.session && this.debug) console.log('Lines added to transaction, but it\'s not commited yet.');
```

Comment:
Lets take this line as an example for `debug` package. Here you could write like this

```javascript
if (options.session) {
  debug('bla bla');
}
```

It will check for the debug flags automatically and you don't need to re-implement the bicycle